src/app/window.cpp

```diff
@@ -60,34 +60,101 @@ void Dash::init()
     auto msg_ref = new QWidget();
     msg_ref->setObjectName("MsgRef");
     this->body.layout->addWidget(msg_ref);
-
+    this->rail.layout->addWidget(this->main_menu());
     this->body.layout->addWidget(this->control_bar());
+    this->set_page(this->arbiter.layout().curr_page);
+}
+
+void Dash::set_page(Page *page)
+{
+    bool fullscreen = this->arbiter.layout().fullscreen_mode;
+    int fullscreen_delay = this->arbiter.layout().fullscreen_delay;
+    if (fullscreen && fullscreen_delay != 0) {
+        this->arbiter.toggle_fullscreen_mode();
+        this->arbiter.toggle_fullscreen_mode();
+    }
+    auto id = this->arbiter.layout().page_id(page);
+    this->rail.group.button(id)->setChecked(true);
+    this->body.frame->setCurrentWidget(page->widget());
+}
+
+QWidget *Dash::main_menu()
+{
+    auto widget = new QWidget();
+    widget->setObjectName("MainMenu");
+    auto layout = new QVBoxLayout(widget);
+    layout->setContentsMargins(0, 0, 0, 0);
+    layout->setSpacing(0);
 
     for (auto page : this->arbiter.layout().pages()) {
-        auto button = page->button();
+        auto button = new QPushButton();
```

Comment:
i think something is still off here... the side buttons arent colored properly

looks like a few more unexpected diffs here (sorry about that :see_no_evil:)